project/npda/forms/visit_form.py

```diff
@@ -740,114 +708,59 @@ def clean_hospital_discharge_date(self):
-                    measurement_method="height",
-                    observation_value=round_to_one_decimal_place(height),
-                    sex=sex,
-                )
-                cleaned_data["height_centile"] = round_to_one_decimal_place(centile)
-                cleaned_data["height_sds"] = round_to_one_decimal_place(sds)
-            except Exception as e:
-                cleaned_data["height_centile"] = None
-                cleaned_data["height_sds"] = None
-                # we are not raising a validation error here as sds and centile are not required fields
-                pass
-        if basic_params and weight is not None:
-            # cleaned_data["weight"] = round_to_one_decimal_place(weight)
-            try:
-                centile, sds = calculate_centiles_z_scores(
-                    birth_date=birth_date,
-                    observation_date=observation_date,
-                    measurement_method="weight",
-                    observation_value=round_to_one_decimal_place(weight),
-                    sex=sex,
-                )
-                cleaned_data["weight_centile"] = round_to_one_decimal_place(centile)
-                cleaned_data["weight_sds"] = round_to_one_decimal_place(sds)
-            except Exception as e:
-                cleaned_data["weight_centile"] = None
-                cleaned_data["weight_sds"] = None
-                # we are not raising a validation error here as sds and centile are not required fields
-                pass
-        # calculate BMI, BMI centile and BMI SDS, if height and weight are present
-        if basic_params and height is not None and weight is not None:
-            cleaned_data["bmi"] = calculate_bmi(height, weight)
-            if (
-                cleaned_data["bmi"] is None
-            ):  # the BMI calculation returns None if BMI is > 99
-                cleaned_data["bmi_centile"] = None
-                cleaned_data["bmi_sds"] = None
-            else:
-                try:
-                    centile, sds = calculate_centiles_z_scores(
-                        birth_date=birth_date,
-                        observation_date=observation_date,
-                        measurement_method="bmi",
-                        observation_value=round(cleaned_data["bmi"], 1),
-                        sex=sex,
-                    )
-                    cleaned_data["bmi_centile"] = round_to_one_decimal_place(centile)
-                    cleaned_data["bmi_sds"] = round_to_one_decimal_place(sds)
-                except Exception as e:
-                    cleaned_data["bmi_centile"] = None
-                    cleaned_data["bmi_sds"] = None
-                    # we are not raising a validation error here as sds and centile are not required fields
-                    pass
+            cleaned_data["weight"] = weight = round_to_one_decimal_place(weight)
+
+        if not getattr(self, "async_validation_results", None):
```

Comment:
This is the same trick we already use for the patient form, allows us to run things asynchronously for the CSV upload but synchronously for the questionnaire form submission